x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/StYMax.java

```diff
@@ -93,37 +105,81 @@ public DataType dataType() {
 
     @Override
     public Expression replaceChildren(List<Expression> newChildren) {
-        return new StYMax(source(), newChildren.get(0));
+        return new StYMax(source(), newChildren.getFirst());
     }
 
     @Override
     protected NodeInfo<? extends Expression> info() {
-        return NodeInfo.create(this, StYMax::new, field());
+        return NodeInfo.create(this, StYMax::new, spatialField());
     }
 
-    @ConvertEvaluator(extraName = "FromWKB", warnExceptions = { IllegalArgumentException.class })
-    static double fromWellKnownBinary(BytesRef wkb) {
-        var geometry = UNSPECIFIED.wkbToGeometry(wkb);
-        if (geometry instanceof Point point) {
-            return point.getY();
+    private static void fromWellKnownBinary(
+        DoubleBlock.Builder results,
+        @Position int p,
+        BytesRefBlock wkbBlock,
+        SpatialEnvelopeVisitor.PointVisitor pointVisitor
+    ) {
+        int firstValueIndex = wkbBlock.getFirstValueIndex(p);
+        int valueCount = wkbBlock.getValueCount(p);
+        if (valueCount == 0) {
+            results.appendNull();
+            return;
         }
-        var envelope = SpatialEnvelopeVisitor.visitCartesian(geometry);
-        if (envelope.isPresent()) {
-            return envelope.get().getMaxY();
+        BytesRef scratch = new BytesRef();
+        var visitor = new SpatialEnvelopeVisitor(pointVisitor);
+        for (int i = 0; i < valueCount; i++) {
+            BytesRef wkb = wkbBlock.getBytesRef(firstValueIndex + i, scratch);
+            var geometry = UNSPECIFIED.wkbToGeometry(wkb);
+            geometry.visit(visitor);
+        }
+        if (pointVisitor.isValid()) {
+            results.appendDouble(visitor.getResult().getMaxY());
+            return;
         }
         throw new IllegalArgumentException("Cannot determine envelope of geometry");
     }
 
-    @ConvertEvaluator(extraName = "FromWKBGeo", warnExceptions = { IllegalArgumentException.class })
```

Comment:
This has also changed from `ConvertEvaluator` to `Evaluator`. What repercussions does that have? Is it a breaking change?